python/llm/src/ipex_llm/transformers/npu_models/llama_mp.py

```diff
@@ -171,13 +172,33 @@ def __init__(
             )
             curr_key_values.append((new_key_states, new_value_states))
 
+        result = hidden_states
+        if model_norm_weight is not None:
+            # fuse norm and lm head
+            model_norm_weight = self.constant(model_norm_weight)
+
+            # norm forward
+            # result = hidden_states
+            result = self.layer_norm(result, model_norm_weight)
+            
+            # lm_head forward
+            result = self.linear(
+                result, 32000, self.hidden_size, bias=False, wt_dtype=self.dtype
+            )
```

Comment:
Maybe move this into `build_decoder` and return linear output instead of `hidden_states`?

Reply:
> Move this into `build_decoder` and return linear output instead of `hidden_states`? Then the output from the decoder layer is the same as before.


yeah, I also tried this but the tensor shape of lm_head output is different from decoder layer, still working on the related send/recv logic to support it.